doc/contribute/committer_expectations.rst

```diff
@@ -0,0 +1,165 @@
+
+Large changes to the Zephyr project must submit an :ref:`RFC proposal <rfcs>`
+describing the full scope of change and future work.  The :ref:`RFC proposal
+<rfcs>` provides the required context to reviewers, but allows for smaller,
+incremental, PRs to get reviewed and merged into the project. The RFC should
+also define the minimum viable implementation.
+
+Changes which require an :ref:`RFC proposal <rfcs>` include:
+
+- Submitting a new feature.
+- Submitting a new API.
+- :ref:`treewide-changes`.
+- Other changes which encompass more than 200 lines of non-trivial modifications.
+
+PR Requirements
+***************
+
+- Each commit in the PR must provide a commit message following the
+  :ref:`commit_message_guidelines`.
+
+- All files in the PR must comply with :ref:`Licensing
+  Requirements<licensing_requirements>`.
+
+- Follow the Zephyr :ref:`coding_style`.
+
+- PRs must pass all CI checks. This is a requirement to merge the PR, but you
+  may mark a PR as draft and explicitly request reviewers to provide early
+  feedback.
+
+- When breaking a PR into multiple commits, each commit must build cleanly. This
+  is not enforced by the CI system, so it is the PR author’s responsibility to
+  verify.
+
+- When major new functionality is added, tests for the new functionality MUST be
+  added to the automated test suite. All new APIs MUST be documented and tested
+  and tests MUST cover at least 80% of the added functionality using the code
+  coverage tool and reporting provided by the project..
+
+- Incompatible changes to APIs must also update the release notes for the
+  next release detailing the change.  APIs marked as experimented are excluded
+  from this requirement. [#api-updates]_
+
+- PRs must also satisfy all :ref:`merge_criteria` before a member of the release
+  engineering team will merge the PR into the zephyr tree.
+
+Maintainers may request you break up a PR into smaller PRs and may request you
+create an :ref:`RFC proposal <rfcs>`.
+
+Workflow Suggestions That Help Reviewers
+========================================
+
+- Don’t resolve comments until the question has been answered or the request
+  completed in an update to the PR.
+
+- Try to minimize rebases in the middle of a review. If a rebase is required,
+  push this as a separate update with no functional changes to the PR. See the
+  note below.
+
+- Instead of amending commits, append fixup commits to the PR to address
+  reviewer comments. This provides reviewers a clear history of the changes made
+  to the PR, along with the rationale for changes. Prior to the final approval
+  and merge of the PR, you must perform a squash merge to cleanup the commit
+  history. (TBD - This needs a PoC to demonstrate this workflow).
+
+Reviewer Expectations
+#####################
+
+- Be respectful when commenting on PRs. Refer to the Zephyr `Code of Conduct`_
+  for more details.
+
+- Review PRs within a reasonable time. (TBD - define soft guidelines for
+  “reasonable”)
```

Comment:
I continue to want to caution you that there are several areas in the project where review bandwidth is extremely underprovisioned. Big examples are the build system and devicetree bindings. We all want prompt review, but we need to also be realistic about the resources we have as an open source project. I continue to be in favor of prompt review for roadmap items approved by the TSC, but there just isn't enough time in all cases to try to do this project-wide IMO.

Reply:
+1. I think a well understood escalation path is probably more valuable than a very fast review SLO.

Reply:
Leaving open for discussion by the Process working group.